hadoop-common-project/hadoop-auth/src/main/java/org/apache/hadoop/util/PlatformName.java

```diff
@@ -45,9 +45,28 @@ public class PlatformName {
 
   /**
    * A public static variable to indicate the current java vendor is
-   * IBM java or not.
+   * IBM and the type is Java Technology Edition which provides its
+   * own implementations of many security packages and Cipher suites.
+   * Note that these are not provided in Semeru runtimes:
+   * See https://developer.ibm.com/languages/java/semeru-runtimes/
    */
-  public static final boolean IBM_JAVA = JAVA_VENDOR_NAME.contains("IBM");
+  public static final boolean IBM_JAVA = shouldUseIbmPackages();
+
+  private static boolean shouldUseIbmPackages() {
+    if (JAVA_VENDOR_NAME.contains("IBM")) {
+      try {
+        /**
+         * This class is provided by all supported IBM JTE Runtimes,
+         * but ensures we do not make assumptions of existence of
+         * specialised security modules based on vendor alone.
```

Comment:
minor nit; you've' used the uk spelling. i'm happy with this, but don't want to confuse others